daemon/cmd/ciliumendpoints.go

```diff
@@ -0,0 +1,107 @@
+	}
+	if enableCiliumEndpointSlice {
+		for _, cesObj := range objs {
+			ces, ok := cesObj.(*cilium_v2a1.CiliumEndpointSlice)
+			if !ok {
+				return fmt.Errorf("unexpected object type returned from ciliumendpointslice store: %T", cesObj)
+			}
+			for _, cep := range ces.Endpoints {
+				if cep.Networking.NodeIP == node.GetCiliumEndpointNodeIP() && eps.LookupPodName(ces.Namespace+"/"+cep.Name) == nil {
+					d.deleteCiliumEndpoint(ctx, ces.Namespace, cep.Name, nil, ciliumClient, eps)
+				}
+			}
+		}
+	} else {
+		for _, cepObj := range objs {
+			cep, ok := cepObj.(*types.CiliumEndpoint)
+			if !ok {
+				return fmt.Errorf("unexpected object type returned from ciliumendpoint store: %T", cepObj)
+			}
+
+			if cep.Networking.NodeIP == node.GetCiliumEndpointNodeIP() && eps.LookupPodName(cep.Namespace+"/"+cep.Name) == nil {
+				d.deleteCiliumEndpoint(ctx, cep.Namespace, cep.Name, &cep.ObjectMeta.UID, ciliumClient, eps)
+			}
+		}
+	}
+	return nil
+}
+
+// deleteCiliumEndpoint safely deletes a CEP by name, if no UID is passed this will reverify that
+// the CEP is still local before doing a delete.
+func (d *Daemon) deleteCiliumEndpoint(
+	ctx context.Context,
+	cepNamespace,
+	cepName string,
+	cepUID *apiTypes.UID,
+	ciliumClient ciliumv2.CiliumV2Interface,
+	eps localEndpointCache) {
+	// To avoid having to store CEP UIDs in CES Endpoints array, we have to get the latest
+	// referenced CEP from apiserver to verify that it still references this node.
+	// To avoid excessive api calls, we only do this if CES is enabled and the CEP
+	// appears to be stale.
+	if cepUID == nil {
+		cep, err := ciliumClient.CiliumEndpoints(cepNamespace).Get(ctx, cepName, metav1.GetOptions{})
+		if err != nil {
+			log.WithError(err).WithFields(logrus.Fields{logfields.CEPName: cepName, logfields.K8sNamespace: cepNamespace}).
+				Error("Failed to get possibly stale CEP from apiserver, skipping.")
+			return
+		}
+		if cep.Status.Networking.NodeIP != node.GetCiliumEndpointNodeIP() {
+			log.WithError(err).WithFields(logrus.Fields{logfields.CEPName: cepName, logfields.K8sNamespace: cepNamespace}).
+				Debug("Stale CEP fetched apiserver no longer references this Node, skipping.")
+			return
+		}
+		cepUID = &cep.ObjectMeta.UID
+	}
+	// There exists a local CiliumEndpoint that is not in the endpoint manager.
+	// This function is run after completing endpoint restoration from local state and K8s cache sync.
+	// Therefore, we can delete the CiliumEndpoint as it is not referencing a Pod that is being managed.
+	// This may occur for various reasons:
+	// * Pod was restarted while Cilium was not running (likely prior to CNI conf being installed).
+	// * Local endpoint was deleted (i.e. due to reboot + temporary filesystem) and Cilium or the Pod where restarted.
+	log.WithFields(logrus.Fields{logfields.CEPName: cepName, logfields.K8sNamespace: cepNamespace}).
+		Info("Found stale ciliumendpoint for local pod that is not being managed, deleting.")
```

Comment:
```suggestion
	log.WithFields(logrus.Fields{logfields.CEPName: cepName, logfields.K8sNamespace: cepNamespace}).
		Info("Found stale CEP for local pod that is not being managed, deleting.")
```

given that the other log msgs in the function already use "CEP".